src/geo/projection/globe_util.js

```diff
@@ -446,21 +455,16 @@ export function calculateGlobeLabelMatrix(tr: Transform, id: CanonicalTileID): F
     // compesanting the reprojection. This has to be compensated for the
     // map aligned label space. Whithout this logic map aligned symbols
     // would appear larger than intended.
-    const m = calculateGlobePosMatrix(x, y, tr.worldSize / tr._projectionScaler, 0, 0);
+    const m = calculateGlobePosMatrix(x, y, tr.worldSize / tr._pixelsPerMercatorPixel, 0, 0);
     return mat4.multiply(m, m, globeDenormalizeECEF(globeTileBounds(id)));
 }
 
 export function calculateGlobeMercatorMatrix(tr: Transform): Float32Array {
-    const worldSize = tr.worldSize;
-    const point = tr.point;
-
-    const mercatorZ = mercatorZfromAltitude(1, tr.center.lat) * worldSize;
-    const projectionScaler = mercatorZ / tr.pixelsPerMeter;
     const zScale = tr.pixelsPerMeter;
-    const ws = worldSize / projectionScaler;
+    const ws = tr.worldSize * tr._pixelsPerMercatorPixel;
```

Comment:
This is the most noteworthy change in this PR, as the value of `ws` now takes into account the interpolation introduced to `transform._projectionScaler `(here renamed to `transform.pixelsPerMercatorPixel`) in [this PR](https://github.com/mapbox/mapbox-gl-js/pull/11951/files#diff-a2b3330768267b8c7996b8bf292fcd864e8b768f538436813b0d7e19f7202176R1690-R1706).

Previously, `ws`varied by latitude at all zoom levels, increasing to `worldSize` at the equator.
After this change, at low zoom levels, `ws` is always the old value at 45 degrees, which is `worldSize * .7` 

I'm not confident about which approach is the correct one, since I'm seeing no difference in behavior, and all tests continue to pass after this change. Does this change in fact make no difference, or are we missing test coverage for an issue?

The previous code can be simplified to:
```js
const ws = tr.pixelsPerMeter / mercatorZfromAltitude(1, tr.center.lat);
```

Reply:
I went ahead and switched this back to the previous algorithm in order to reduce the chance of a regression.